scrapi/harvesters/vivo.py

```diff
@@ -0,0 +1,238 @@
+            variables += '?' + variable + ' '
+        pattern = sparql_map['pattern'].format(uri)
+        sparql_query = self.QUERY_TEMPLATE.format(variables, pattern)
+        self.sparql_wrapper.setQuery(sparql_query)
+        results = self.sparql_wrapper.query()
+        results = results.convert()
+        ret = []
+        for result in results['results']['bindings']:
+            item = {}
+            for variable in sparql_map['fields']:
+                item[variable] = result[variable]['value']
+            ret.append(item)
+        return ret
+
+    def get_records(self, uris):
+        records = []
+        for uri in uris:
+            record = {}
+            record['uri'] = uri
+            for sparql_map in SPARQL_MAPPING:
+                if sparql_map['type'] == 'string':
+                    record[sparql_map['name']] = self.get_string(uri, sparql_map)
+                if sparql_map['type'] == 'array':
+                    record[sparql_map['name']] = self.get_array(uri, sparql_map)
+                if sparql_map['type'] == 'dict':
+                    record[sparql_map['name']] = self.get_dict(uri, sparql_map)
+            records.append(record)
+        return records
+
+    def get_total(self, start_date, end_date):
+        query_str = """PREFIX vivo:  <http://vivoweb.org/ontology/core#>
+                       PREFIX xsd:   <http://www.w3.org/2001/XMLSchema#>
+                       SELECT (COUNT(*) AS ?total)
+                       {{
+                            ?s vivo:dateTimeValue ?dateURI .
+                            ?dateURI vivo:dateTime ?date .
+                            FILTER (strdt(?date, xsd:date) >= "{}"^^xsd:date && strdt(?date, xsd:date) <= "{}"^^xsd:date)
+                        }}"""
+        query_str = query_str.format(start_date.isoformat(), end_date.isoformat())
+        self.sparql_wrapper.setQuery(query_str)
+        result = self.sparql_wrapper.query()
+        result = result.convert()
+        return int(result['results']['bindings'][0]['total']['value'])
+
+    def get_uris(self, start_date, end_date, limit, offset):
+        uris = []
+        query_str = """ PREFIX vivo:  <http://vivoweb.org/ontology/core#>
+                        PREFIX xsd:   <http://www.w3.org/2001/XMLSchema#>
+
+                        SELECT ?uri
+                        {{
+                            ?uri vivo:dateTimeValue ?dateUri .
+                            ?dateUri vivo:dateTime ?date .
+                            FILTER (strdt(?date, xsd:date) >= "{}"^^xsd:date && strdt(?date, xsd:date) <= "{}"^^xsd:date)
+                        }}  LIMIT {} OFFSET {}
+                    """
+        query_str = query_str.format(start_date.isoformat(), end_date.isoformat(), limit, offset)
+        self.sparql_wrapper.setQuery(query_str)
+        results = self.sparql_wrapper.query()
+        results = results.convert()
+        for result in results['results']['bindings']:
```

Comment:
you can just 
```
return [ result['uri']['value']  for result in results['results']['bindings']]
```
if you want, but this is really minor stylistic quibble, so feel free to ignore.